modules/benchmark/R/metric_lmDiag_plot.R

```diff
@@ -7,9 +7,9 @@
 metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE) {
   PEcAn.logger::logger.info("Metric: Linear Regression Diagnostic Plot")
   
-  fit <- lm(metric_dat[, 1] ~ metric_dat[, 2])
+  fit <- stats::lm(metric_dat[, 1] ~ metric_dat[, 2])
 
-  p1 <- ggplot2::ggplot(fit, aes(.fitted, .resid))
+  p1 <- ggplot2::ggplot(fit, ggplot2::aes(.fitted, .resid))
```

Comment:
```suggestion
  p1 <- ggplot2::ggplot(fit, ggplot2::aes(.data$.fitted, .data$.resid))
```
Here's another sneaky one: Where did `.fitted` and `.resid` come from and how do we keep R CMD check from complaining about them? Well! 
<img width="925" alt="Screenshot 2024-02-13 at 1 34 06 PM" src="https://github.com/PecanProject/pecan/assets/1256435/63d71929-455b-40c7-a99e-dd579861a8ca">
It turns out ggplot is converting the `fit` object into a data frame under the hood using `fortify()`, so we can pass any of the names `fortify.lm` creates into `.data` just as if they were already present in the dataframe.
<img width="643" alt="Screenshot 2024-02-13 at 1 35 10 PM" src="https://github.com/PecanProject/pecan/assets/1256435/6c9a0a29-5bef-478e-a217-234fe1571146">

Reply:
(This feels like a lot of magic to rely on here, but when trying to make the package checks quiet down I think simply prefixing `.data$` on what's already here is the right call)